### PR TITLE
document popolo data model

### PR DESCRIPTION
**Pull request by struan**
Pages to document how to use the data model by using some common examples.

Also a tutorial on using the API and a page listing some of the client libraries available.

### CODE REVIEW

File: docs/api-tutorial.md [hunk 1]
```diff
@@ -0,0 +1,317 @@
...
+      }
+    }
+
+{% endhighlight %}
+
+Note that the `--data` line is now
+
+       --data '{"id": "joe-bloggs", "name": "Joe Bloggs"}'
+
+and that the `id` field and the two URL are different. If we'd tried to use an existing ID Popit would have returned an error.
```
Comment (tmtmtmtm) [inline docs/api-tutorial.md]: two URLs

File: docs/api-tutorial.md [hunk 2]
```diff
@@ -0,0 +1,317 @@
...
+    }
+  }
+
+{% endhighlight %}
+
+( The error message above has been slightly edited to make it easier to read )
+
+Of course, a list of people on their own is only so useful, we need to add organizations and link those with people.
+
+### Creating Organization ###
```
Comment (tmtmtmtm) [inline docs/api-tutorial.md]: Creating _an_ Organization

[might also be worth being consistent between Adding and Creating]

File: docs/api-tutorial.md [hunk 3]
```diff
@@ -0,0 +1,317 @@
...
+      "contact_details": [],
+      "identifiers": [],
+      "other_names": [],
+      "url": "http://walkthrough.popit.mysociety.org/api/v0.1/persons/joe-bloggs",
+      "html_url": "http://walkthrough.popit.mysociety.org/persons/joe-bloggs"
+    }
+  }
+{% endhighlight %}
+
+Note that we have to include the name in each request, but we only need to include the information we want to change. If you don't include a field in the update then it will remain unchanged:
```
Comment (tmtmtmtm) [inline docs/api-tutorial.md]: This differs from the docs at http://popit.poplus.org/docs/api/writing/, which claim "Updating a document is done by sending the whole document in the PUT request. Any fields that are omitted in the request will be removed, and replaced with the default."

File: docs/using-the-popolo-data-model.md [hunk 4]
```diff
@@ -0,0 +1,170 @@
...
+
+An organization is used to represent something to which people belong such as a parliament, political party or a company.
+
+#### Memberships ####
+
+A membership represents an association between two things. Mostly you'll use this to indicate that a person is a member of an organization.
+
+#### Posts ####
+
+A post is for the sort of position that it's useful to represent independent of the organization it's part of or the person who holds it. Unless you are modeling a parliament that has constituency seats ( e.g. Member of Parliament for Oxford East ) then you probably won't need posts.
```
Comment (tmtmtmtm) [inline docs/using-the-popolo-data-model.md]: At this point the document hasn't said yet that it's primarily about modelling Parliaments, so this claim seems a little over-broad.

File: docs/using-the-popolo-data-model.md [hunk 5]
```diff
@@ -0,0 +1,170 @@
...
+
+### But what about roles? ###
+
+It's possible for a membership, or a post, to have a role. A role is used to indicate something about the nature of the membership. E.g. for people in a party you could set the role of the membership to be member. Or for people on a partly list the role could be candidate.
+
```
Comment (tmtmtmtm) [inline docs/using-the-popolo-data-model.md:36]: partly → party

I fin this paragraph quite hard to follow — I suspect it would be largely impenetrable for someone looking at this with no understanding already. I think an example here showing a Membership (and/or Post) without a Role, and then with one, could get the idea across well. In general, for a document like this (as opposed to a reference guide) the key issue is showing _why_ you would or wouldn't use something, beyond just what it is.

File: docs/using-the-popolo-data-model.md (hunk 4 above)
Comment (tmtmtmtm) [inline docs/using-the-popolo-data-model.md]: I think a useful concept to get across here is that Posts are useful when you have a position that can exist even if someone doesn't currently hold it.

File: docs/using-the-popolo-data-model.md [hunk 6]
```diff
@@ -0,0 +1,170 @@
...
+        "label": "Member",
+        "start_date": "2009-05-06",
+        "links": [],
+        "contact_details": []
+      }
+    ]
+  }
+}
+{% endhighlight %}
+
```
Comment (tmtmtmtm) [inline docs/using-the-popolo-data-model.md:69]: I'd probably leave out the fields here that aren't important to the example — links, contact_details, possibly even url

File: docs/using-the-popolo-data-model.md [hunk 7]
```diff
@@ -0,0 +1,170 @@
...
+  "url": "http://za-peoples-assembly.popit.mysociety.org/api/v0.1/memberships/org.mysociety.za/membership/743",
+  "id": "org.mysociety.za/membership/743",
+  "organization_id": "org.mysociety.za/house/national-assembly",
+  "role": "Member",
+  "person_id": "org.mysociety.za/person/104",
+  "label": "Member",
+  "start_date": "2009-05-06",
+  "area": {
+    "name": "Area Name"
+  },
```
Comment (tmtmtmtm) [inline docs/using-the-popolo-data-model.md]: It's slightly confusing to me why for this sort of example the area just gets a name, but in the later one gets or needs Popit identifiers too.

File: docs/using-the-popolo-data-model.md [hunk 8]
```diff
@@ -0,0 +1,170 @@
...
+  "links": [],
+  "contact_details": []
+}
+{% endhighlight %}
+
+#### Constituency based parliament ####
+
+In this model there are a number of constituencies in which one member is elected. In this case we create a post for each constituency in the parliament. The elected representatives are then associated with the post using a membership.
```
Comment (tmtmtmtm) [inline docs/using-the-popolo-data-model.md]: Is this saying this is the only sensible way to do this; or there are multiple ways, but we think this is probably best; or just this is one good way to do it?

File: docs/using-the-popolo-data-model.md [hunk 9]
```diff
@@ -0,0 +1,170 @@
...
+      "person_id": "david-cameron",
+      "post_id": "65622",
+      "role": "Member",
+      "start_date": "2005-05-06",
+      "end_date": "2010-05-06",
+      "links": [],
+      "contact_details": [],
+    }
+  ],
+}
```
Comment (tmtmtmtm) [inline docs/using-the-popolo-data-model.md:113]: I think leaving out the unnecessary fields (and here I'd include 'id' as well) would definitely make this one a lot clearer.

File: docs/using-the-popolo-data-model.md [hunk 10]
```diff
@@ -0,0 +1,170 @@
...
+      "id": "5448efe57f292c7a67b292ab",
+      "person_id": "david-cameron",
+      "post_id": "65622",
+      "role": "Member",
+      "start_date": "2005-05-06",
+      "end_date": "2010-05-06",
+      "links": [],
+      "contact_details": [],
+    },
+  ],
```
Comment (tmtmtmtm) [inline docs/using-the-popolo-data-model.md]: Including David Cameron's membership confuses me, as we're showing this in the context of what the Post referenced from the David Cameron record should be, so it seems circular. I think there's a mismatch here between showing what data you'll use to _create_ these records, and what the records will look like, in context, once they've been connected. For most new users I suspect the first (ie what they'll personally need to create) is much more important than the latter.

File: docs/api-tutorial.md (hunk 3 above)
Comment (mhl) [inline docs/api-tutorial.md]: This is confusing at the moment - see also: https://github.com/mysociety/popit-api/issues/95

File: docs/api-tutorial.md [hunk 11]
```diff
@@ -0,0 +1,316 @@
+---
+layout: docs
+title: API tutorial
+---
+
+This tutorial aims to show you the basics of using the API. For simplicity we're going to use the command line tool `curl` although for use as part of a larger project you'll probably use either the appropriate [PopIt client library](/docs/api/clients/) or an HTTP client.
+
+We're going to assume that you've got an account and an API key, as described in the Authentication section of [Using the API ](/docs/api/).
```
Comment (chrismytton) [inline docs/api-tutorial.md]: Might be good to link directly to the authentication section with `/docs/api/#authentication`

File: docs/api-tutorial.md [hunk 12]
```diff
@@ -0,0 +1,316 @@
...
+
+### Getting the API URL ###
+
+Firstly we need the URL for the API for your instance. If you go to the front page of your instance you should see a textbox next to the description:
+
+<img src="{{ site.baseurl }}/assets/img/walkthrough-api-url-location.png" />
+
+The URL in this textbox is the base URL for your instance's API.
+
+For these examples we're going to assume our instance is called walkthroug and its API is accessed at `https://walkthrough.popit.mysociety.org/api/v0.1`.
```
Comment (chrismytton) [inline docs/api-tutorial.md]: Typo "walkthroug"

File: docs/api-tutorial.md [hunk 13]
```diff
@@ -0,0 +1,316 @@
...
+
+As you can see it returns a list of the API URLs used to access the various different objects that PopIt uses. For more details about what each of these are see the guide to [using the Popolo data model](/docs/using-the-popolo-data-model/).
+
+### Getting data ###
+
+Getting data out of PopIt using the API is simple. To get a list of all the items of a certain type you just call the relevant API end point we saw in the call above. So, to get details of all the people stored in our instance we call `https://walkthrough.popit.mysociety.org/api/v0.1/persons`:
+
+{% highlight bash %}
+
+https://walkthrough.popit.mysociety.org/api/v0.1/persons/
```
Comment (chrismytton) [inline docs/api-tutorial.md]: I think it would make sense for this to be prefixed with `$ curl` like the other examples in this page.

**Comment by chrismytton** [comment]
Would be good to make all the urls consistently use `https://`.

File: docs/api-tutorial.md [hunk 14]
```diff
@@ -0,0 +1,316 @@
...
+
+{% endhighlight %}
+
+Note that the `--data` line is now
+
+       --data '{"id": "joe-bloggs", "name": "Joe Bloggs"}'
+
+and that the `id` field and the two URLs are different. If we'd tried to use an existing ID PopIt would have returned an error.
+
+{% highlight json %}
```
Comment (chrismytton) [inline docs/api-tutorial.md]: I think this needs to be `highlight bash`, otherwise the highlighting goes a bit wacky!

File: docs/using-the-popolo-data-model.md [hunk 15]
```diff
@@ -0,0 +1,133 @@
...
+PopIt uses the [Popolo specification](http://www.popoloproject.com/) to model people, organizations and their relationships. Understanding the best way to use this to model your needs is key to getting the best out of PopIt. This page aims to explain to you the basic concepts of Popolo and how to use them in some common scenarios that come up when modeling parliaments.
+
+### Basic concepts ###
+
+Popolo, and hence PopIt, data models are based on a few key objects:
+
+   * Persons
+   * Organizations
+   * Memberships
+   * Posts
```
Comment (chrismytton) [inline docs/using-the-popolo-data-model.md]: To you think it's worth linking these to the relevant popolo spec page?

File: docs/using-the-popolo-data-model.md [hunk 16]
```diff
@@ -0,0 +1,133 @@
...
+        "person_id": "org.mysociety.za/person/104",
+      }
+    ]
+  }
+}
+{% endhighlight %}
+
+#### Party lists for geographic areas ####
+
+In this case candidates are still elected proportionally from party lists but the lists are specified per area. E.g. There would be a set of party lists for South West England and multiple people would be elected for this area. In this case the model is the same are previously with the addition that when creating the memberships you assign an area to the membership.
```
Comment (chrismytton) [inline docs/using-the-popolo-data-model.md]: s/is the same _are_ previously/is the same _as_ previously

**Comment by chrismytton** [comment]
Few minor comments but overall this is really great! :+1:
